networkx/algorithms/centrality/tests/test_current_flow_closeness.py

```diff
@@ -32,10 +34,42 @@ def test_star(self):
         for n in sorted(G):
             assert b[n] == pytest.approx(b_answer[n], abs=1e-7)
 
-    def test_current_flow_closeness_centrality_not_connected(self):
-        G = nx.Graph()
-        G.add_nodes_from([1, 2, 3])
-        with pytest.raises(nx.NetworkXError):
+    @pytest.mark.parametrize(
+        ("G", "expectation"),
+        [
+            (
+                nx.Graph(),
+                pytest.raises(
+                    nx.NetworkXPointlessConcept,
+                    match="centrality is undefined for the null graph",
+                ),
+            ),
+            (
+                nx.path_graph(1),
+                pytest.raises(
+                    nx.NetworkXError,
+                    match="graph with 1 node has fewer than three nodes",
+                ),
+            ),
+            (
+                nx.Graph([(0, 1)]),
+                pytest.raises(
+                    nx.NetworkXError,
+                    match="graph with 2 nodes has fewer than three nodes",
+                ),
+            ),
+            (nx.Graph([(0, 1), (1, 2)]), does_not_raise()),
+            (
+                nx.Graph([(0, 1), (2, 3)]),
+                pytest.raises(nx.NetworkXError, match="graph is not connected"),
+            ),
+            (nx.DiGraph(), pytest.raises(nx.NetworkXNotImplemented)),
+            (nx.MultiGraph(), pytest.raises(nx.NetworkXNotImplemented)),
+            (nx.MultiDiGraph(), pytest.raises(nx.NetworkXNotImplemented)),
+        ],
+    )
+    def test_current_flow_closeness_centrality_exceptions(self, G, expectation):
+        with expectation:
             nx.current_flow_closeness_centrality(G)
```

Comment:
you can probably break this down into 3 tests to make it more understandable.
Also, The MultiGraph and MultiDiGraph tests will be added after the deprecation warning will be removed. Right now we will have a test to check if we are getting the deprecation warning for multigraphs

```suggestion
    @pytest.mark.parametrize(
        ("G", "expectation", "msg"),
        [
            (
                nx.Graph(),
                nx.NetworkXPointlessConcept,
                "centrality is undefined for the null graph",
            ),
            (
                nx.path_graph(1),
                nx.NetworkXError,
                "graph with 1 node has fewer than three nodes",
            ),
            (
                nx.Graph([(0, 1)]),
                nx.NetworkXError,
                "graph with 2 nodes has fewer than three nodes",
            ),
            (
                nx.Graph([(0, 1), (2, 3)]),
                nx.NetworkXError,
                "graph is not connected",
            ),
        ],
    )
    def test_current_flow_closeness_centrality_exceptions(self, G, expectation, msg):
        with pytest.raises(exception, match=msg):
            nx.current_flow_closeness_centrality(G)

    @pytest.mark.parametrize("G", (nx.DiGraph(), nx.MultiGraph(), nx.MultiDiGraph()))
    def test_current_flow_closeness_centrality_not_implemented_for(self):
        with pytest.raises(nx.NetworkXNotImplemented):
            nx.current_flow_closeness_centrality(G)

    def test_current_flow_closeness_centrality():
        G = nx.Graph([(0, 1), (1, 2)])
        assert nx.current_flow_closeness_centrality(G) == # output

```

Reply:
I agree breaking it down will make things more readable - you showed me the parametrized tests earlier and I've gone completely overboard with them since...

Re: the deprecation warnings, are we certain now that they will be required? I'm going to hold off on changing that part until I know for sure what should happen to them.